core/src/main/java/hu/uni/miskolc/iit/softwaretesting/model/Book.java

```diff
@@ -29,8 +33,15 @@
     private Genre genre;
 
 
-    public Book(String author, String title, Long isbn, int publishDate, Genre genre)
-    {
+    public Book(String author, String title, Long isbn, int publishDate, Genre genre) {
+
+        if (Calendar.getInstance().get(Calendar.YEAR)< publishDate)
```

Comment:
You can put these conditions to a separate method to avoid code redundancy (at the moment, it uses the same lines of code in the constructor and the setPublishDate method).